src/bin/stratis_uuids_to_names.rs

```diff
@@ -0,0 +1,206 @@
+
+use std::{
+    collections::HashMap,
+    env::args,
+    error::Error,
+    fmt::{self, Debug, Display},
+    os::unix::net::UnixDatagram,
+    time::Duration,
+};
+
+use dbus::{
+    arg::{RefArg, Variant},
+    blocking::Connection,
+    Path,
+};
+use lazy_static::lazy_static;
+use regex::Regex;
+use uuid::Uuid;
+
+pub const DEV_LOG: &str = "/dev/log";
+/// Syslog priority syntax
+/// (3 (SYSTEM) << 3) | 6 (INFO)
+pub const SYSTEM_DAEMON_INFO: &str = "<30>";
+/// Syslog priority syntax
+/// (3 (SYSTEM) << 3) | 3 (ERROR)
+pub const SYSTEM_DAEMON_ERROR: &str = "<27>";
+pub const STRATIS_BUS_NAME: &str = "org.storage.stratis2";
+pub const STRATIS_MANAGER_OBJECT: &str = "/org/storage/stratis2";
+pub const STRATIS_POOL_IFACE: &str = "org.storage.stratis2.pool.r1";
+pub const STRATIS_FS_IFACE: &str = "org.storage.stratis2.filesystem";
+pub const DBUS_OM_IFACE: &str = "org.freedesktop.DBus.ObjectManager";
+lazy_static! {
+    static ref TIMEOUT: Duration = Duration::new(5, 0);
+}
+
+struct StratisUdevError(Option<String>);
+
+impl StratisUdevError {
+    fn empty() -> StratisUdevError {
+        StratisUdevError(None)
+    }
+
+    fn new<D>(display: D) -> StratisUdevError
+    where
+        D: Display,
+    {
+        StratisUdevError(Some(display.to_string()))
+    }
+}
+
+impl<E> From<E> for StratisUdevError
+where
+    E: Error,
+{
+    fn from(e: E) -> StratisUdevError {
+        StratisUdevError::new(e)
+    }
+}
+
+impl Debug for StratisUdevError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "Could not convert Stratis UUIDs to names.")?;
+        if let Some(ref msg) = self.0 {
+            write!(f, " {}", msg)
+        } else {
+            Ok(())
+        }
+    }
+}
+
+type GMORet =
+    HashMap<Path<'static>, HashMap<String, HashMap<String, Variant<Box<dyn RefArg + 'static>>>>>;
+
+fn get_managed_objects() -> Result<GMORet, StratisUdevError> {
+    let connection = Connection::new_system()?;
+    let proxy = connection.with_proxy(STRATIS_BUS_NAME, STRATIS_MANAGER_OBJECT, *TIMEOUT);
+    Ok(proxy
+        .method_call(DBUS_OM_IFACE, "GetManagedObjects", ())
+        .map(|r: (GMORet,)| r.0)?)
+}
+
+fn udev_name_to_uuids(dm_name: &str) -> Result<Option<(Uuid, Uuid)>, StratisUdevError> {
+    let regex = Regex::new("stratis-1-(?P<pool>[0-9a-f]{32})-thin-fs-(?P<fs>[0-9a-f]{32})")?;
+    let captures = match regex.captures(dm_name) {
+        Some(cap) => cap,
+        None => return Ok(None),
+    };
+    let pool_uuid = captures
+        .name("pool")
+        .map(|pu| Uuid::parse_str(pu.as_str()).expect("Format validated by regex"));
+    let fs_uuid = captures
+        .name("fs")
+        .map(|pu| Uuid::parse_str(pu.as_str()).expect("Format validated by regex"));
+    Ok(pool_uuid.and_then(|pu| fs_uuid.map(|fu| (pu, fu))))
+}
+
+fn uuid_to_stratis_name(
+    managed_objects: &GMORet,
+    iface_name: &'static str,
+    uuid: Uuid,
+) -> Result<Option<String>, StratisUdevError> {
```

Comment:
I think ```Option``` part of return type is unnecessary; either there is an error or the name is found.